shared/fs/folder/container.js

```diff
@@ -5,12 +5,11 @@ import Folder from '.'
 import * as Types from '../../constants/types/fs'
 import * as Constants from '../../constants/fs'
 import * as FsGen from '../../actions/fs-gen'
-import flags from '../../util/feature-flags'
 
 const mapStateToProps = (state, {path}) => ({
   _pathItem: state.fs.pathItems.get(path, Constants.unknownPathItem),
   _username: state.config.username,
-  offline: !state.fs.kbfsDaemonStatus.online,
+  offline: Constants.isOfflineUnsynced(state.fs.kbfsDaemonStatus, state.fs.pathItems, path),
```

Comment:
We should try to do as less as possible in `mapStateToProps` since any update to the redux store needs to iterate through all of these connected components. This is fast so it probably won't make a huge difference, but you already have `_pathItem` above; why now have `kbfsDaemonStatus` here and compute `isOfflineUnsyunced` in `mergeProps`?

Reply:
I considered that - the reason I went with computing it here is that `Constants.isOfflineUnsynced` is called elsewhere, from somewhere we don't have the `pathItem`. The alternative that I had thought of, which seemed worse, was to return all the `pathItems` from `mapStateToProps`, but I elected not to do that since it would make the component render too often.

As a third alternative, I could change `Constants.isOfflineUnsynced` to take a `pathItem` and make the sort bar get the path item when it needs it. I'll do that.

Reply:
> As a third alternative, I could change `Constants.isOfflineUnsynced` to take a pathItem and make the sort bar get the path item when it needs it. I'll do that.

Yeah this sounds like the right approach. `Constants.isOfflineUnsynced` only needs a single `pathItem`, so it should just take that instead of `state.fs.pathItems`.